modules/stochastic_tools/src/surrogates/PolynomialRegressionTrainer.C

```diff
@@ -107,35 +138,46 @@ PolynomialRegressionTrainer::train()
 
   // Emplace new values if necessary
   if (_rvecval)
+  {
     for (unsigned int r = _rhs.size(); r < _rvecval->size(); ++r)
+    {
       _rhs.emplace_back(_n_poly_terms, 0.0);
+      _r_sum.emplace_back(0.0);
+    }
+  }
 
   for (unsigned int i = 0; i < _n_poly_terms; ++i)
   {
-    std::vector<unsigned int> i_powers(_power_matrix[i]);
-
     Real i_value(1.0);
     for (unsigned int ii = 0; ii < _n_dims; ++ii)
-      i_value *= data_pow(ii, i_powers[ii]);
+      i_value *= data_pow(ii, _power_matrix[i][ii]);
 
     for (unsigned int j = 0; j < _n_poly_terms; ++j)
     {
-      std::vector<unsigned int> j_powers(_power_matrix[j]);
-
       Real j_value(1.0);
       for (unsigned int jj = 0; jj < _n_dims; ++jj)
-        j_value *= data_pow(jj, j_powers[jj]);
+        j_value *= data_pow(jj, _power_matrix[j][jj]);
 
       _matrix(i, j) += i_value * j_value;
     }
 
+    // Update calculators.
+    for (auto & calc_type : _calculators)
+      calc_type.second[i]->updateCalculator(i_value);
```

Comment:
```suggestion
    for (unsigned int c = i * 3; c < (i + 1) * 3; ++c)
      _calculators[i]->updateCalculator(i_value);
```

Reply:
This should be `_calculators[c]`, right?